operators/pkg/forge/resourcequota_test.go

```diff
@@ -0,0 +1,156 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package forge_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	clv1alpha1 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha1"
+	clv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/forge"
+)
+
+var _ = Describe("Resource quota spec forging", func() {
+
+	Describe("The forge.TenantResourceList function", func() {
+		var (
+			tnQuotas   clv1alpha2.TenantResourceQuota
+			workspaces []clv1alpha1.Workspace
+		)
+
+		BeforeEach(func() {
+			workspaces = make([]clv1alpha1.Workspace, 0)
+			var sampleWorkspace1, sampleWorkspace2 clv1alpha1.Workspace
+			// sample resource quota spec for each workspace
+			quotas1 := clv1alpha1.WorkspaceResourceQuota{
+				CPU:       *resource.NewQuantity(10, resource.DecimalSI),
+				Memory:    *resource.NewScaledQuantity(15, resource.Giga),
+				Instances: 2,
+			}
+
+			quotas2 := clv1alpha1.WorkspaceResourceQuota{
+				CPU:       *resource.NewQuantity(20, resource.DecimalSI),
+				Memory:    *resource.NewScaledQuantity(25, resource.Giga),
+				Instances: 3,
+			}
+			sampleWorkspace1.Spec.Quotas = quotas1
+			sampleWorkspace2.Spec.Quotas = quotas2
+			workspaces = append(workspaces, sampleWorkspace1, sampleWorkspace2)
+		})
+
+		JustBeforeEach(func() {
+			tnQuotas = forge.TenantResourceList(workspaces)
+		})
+
+		When("Forging the tenant resource quota", func() {
+			It("Should have total amount of CPU equal to the sum for each workspace", func() {
+				Expect(tnQuotas.CPU).To(Equal(*resource.NewQuantity(30, resource.DecimalSI)))
+			})
+
+			It("Should have total amount of memory equal to the sum for each workspace", func() {
+				Expect(tnQuotas.Memory).To(Equal(*resource.NewScaledQuantity(40, resource.Giga)))
+			})
+
+			It("Should have total number of instances equal to the sum for each workspace", func() {
+				Expect(tnQuotas.Instances).To(Equal(int64(5)))
+			})
+		})
+	})
+
+	Describe("The forge.ResourceQuotaSpec function", func() {
+		var (
+			spec   corev1.ResourceList
+			tenant clv1alpha2.Tenant
+		)
+
+		Describe("Forging the resource quota specifications", func() {
+
+			BeforeEach(func() {
+				tenant = clv1alpha2.Tenant{}
+				// sample resource quota associated with a Tenant
+				tnQuotasForSpec := clv1alpha2.TenantResourceQuota{
+					CPU:       *resource.NewQuantity(25, resource.DecimalSI),
+					Memory:    *resource.NewScaledQuantity(50, resource.Giga),
+					Instances: 6,
+				}
+
+				tnQuotasForStatus := clv1alpha2.TenantResourceQuota{
+					CPU:       *resource.NewQuantity(20, resource.DecimalSI),
+					Memory:    *resource.NewScaledQuantity(40, resource.Giga),
+					Instances: 5,
+				}
+				tenant.Spec.Quotas = tnQuotasForSpec
+				tenant.Status.Quotas = tnQuotasForStatus
+			})
```

Comment:
Should construct the tests appropriately to avoid code duplication. Please, have a look at the instance controller tests for examples on how to do that.